manta-pay/src/crypto/hash/poseidon/parameter_generation/lfsr.rs

```diff
@@ -0,0 +1,195 @@
+            30,
+            0b111111111111111111111111111111u128,
+        );
+        let mut res = GrainLFSR {
+            state: init_sequence,
+            prime_num_bits,
+            head,
+        };
+        res.init();
+        res
+    }
+
+    fn update(&mut self) -> bool {
+        let new_bit =
+            self.bit(62) ^ self.bit(51) ^ self.bit(38) ^ self.bit(23) ^ self.bit(13) ^ self.bit(0);
+        self.state[self.head] = new_bit;
+        self.head += 1;
+        self.head %= 80;
+        new_bit
+    }
+
+    fn init(&mut self) {
+        for _ in 0..160 {
+            self.update();
+        }
+    }
+
+    #[inline]
+    fn bit(&self, index: usize) -> bool {
+        self.state[(index + self.head) % 80]
+    }
+
+    pub fn get_bits(&mut self, num_bits: usize) -> Vec<bool> {
+        let mut res = Vec::new();
+
+        for _ in 0..num_bits {
+            // Obtains the first bit
+            let mut new_bit = self.update();
+
+            // Loop until the first bit is true
+            while !new_bit {
+                // Discards the second bit
+                let _ = self.update();
+                // Obtains another first bit
+                new_bit = self.update();
+            }
+
+            // Obtains the second bit
+            res.push(self.update());
+        }
+
+        res
+    }
+
+    pub fn get_field_elements_rejection_sampling<F>(&mut self, num_elems: usize) -> Vec<F>
+    where
+        F: ParamField,
+    {
+        assert_eq!(F::MODULUS_BITS as u64, self.prime_num_bits);
+
+        let mut res = Vec::new();
+        for _ in 0..num_elems {
+            // Performs rejection sampling
+            loop {
+                // Obtain n bits and make it most-significant-bit first
+                let mut bits = self.get_bits(self.prime_num_bits as usize);
+                bits.reverse();
+
+                // Constructs the number
+                if let Some(f) = F::try_from_bits_le(&bits) {
+                    res.push(f);
+                    break;
+                }
+            }
+        }
+
+        res
+    }
+
+    pub fn get_field_elements_mod_p<F>(&mut self, num_elems: usize) -> Vec<F>
+    where
+        F: ParamField,
+    {
+        assert_eq!(F::MODULUS_BITS as u64, self.prime_num_bits);
+
+        let mut res = Vec::new();
+        for _ in 0..num_elems {
+            // Obtains n bits and make it most-significant-bit first
+            let mut bits = self.get_bits(self.prime_num_bits as usize);
+            bits.reverse();
+
+            let bytes = bits
+                .chunks(8)
+                .map(|chunk| {
+                    let mut result = 0u8;
+                    for (i, bit) in chunk.iter().enumerate() {
+                        result |= u8::from(*bit) << i
+                    }
+                    result
+                })
+                .collect::<Vec<u8>>();
+
+            res.push(F::from_le_bytes_mod_order(&bytes));
+        }
+
+        res

```

Comment:
```suggestion
        assert_eq!(F::MODULUS_BITS as u64, self.prime_num_bits);
        let mut field_elements = Vec::new();
        for _ in 0..num_elems {
            let mut bits = self.get_bits(self.prime_num_bits as usize);
            bits.reverse();
            let bytes = bits
                .chunks(8)
                .map(|chunk| {
                    let mut result = 0u8;
                    for (i, bit) in chunk.iter().enumerate() {
                        result |= u8::from(*bit) << i
                    }
                    result
                })
                .collect::<Vec<u8>>();
            field_elements.push(F::from_le_bytes_mod_order(&bytes));
        }
        field_elements
```

Reply:
Separate 

```rust
let mut result = 0u8;
for (i, bit) in chunk.iter().enumerate() {
    result |= u8::from(*bit) << i
}
result
```

into its own function. Also, remember to use relevant names where possible: `result` doesn't describe what it is.